src/robot/parsing/model/statements.py

```diff
@@ -971,26 +971,35 @@ class WhileHeader(Statement):
     type = Token.WHILE
 
     @classmethod
-    def from_params(cls, condition, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+    def from_params(cls, condition, limit=None, indent=FOUR_SPACES,
+                    separator=FOUR_SPACES, eol=EOL):
         tokens = [Token(Token.SEPARATOR, indent),
                   Token(cls.type),
                   Token(Token.SEPARATOR, separator),
-                  Token(Token.ARGUMENT, condition),
-                  Token(Token.EOL, eol)]
+                  Token(Token.ARGUMENT, condition)]
+        if limit:
+            tokens.extend([Token(Token.SEPARATOR, indent),
+                           Token(Token.OPTION, limit)])
+        tokens.append(Token(Token.EOL, eol))
         return cls(tokens)
 
     @property
     def condition(self):
-        values = self.get_values(Token.ARGUMENT)
-        if len(values) != 1:
-            return ', '.join(values) if values else None
-        return values[0]
+        return ', '.join(self.get_values(Token.ARGUMENT))
```

Comment:
This changes old behavior so that if there's no condition, an empty string is returned instead of `None`. I guess that's fine because `validate` detects the error situation anyway.